public/jsx/views/complete-payment.jsx

```diff
@@ -16,12 +19,26 @@ module.exports = React.createClass({
     };
   },
 
+  handleClick: function(e) {
+    e.preventDefault();
+    if (window.parent !== window) {
+      // Note: the targetOrigin is wide open.
+      // Nothing sensitive should be sent whilst
+      // that's the case.
+      console.log('Telling parent to close modal.');
+      window.parent.postMessage('close', '*');
+    } else {
+      console.log('Not iframed. No-op');
```

Comment:
I have a feeling one day we might support a redirect flow for mobile. We'll figure it out later.